R/data-combined.R

```diff
@@ -0,0 +1,860 @@
+      names <- validateVectorArgs(names, type = "character")
+
+      # apply specified data transformations
+      private$.dataCombined <- private$.dataTransform(
+        data          = private$.dataCombined,
+        names         = names,
+        xOffsets      = xOffsets,
+        yOffsets      = yOffsets,
+        xScaleFactors = xScaleFactors,
+        yScaleFactors = yScaleFactors
+      )
+
+      # update private field with transformation values
+      private$.dataTransformations <- private$.extractTransforms(private$.dataCombined)
+
+      # for method chaining
+      invisible(self)
+    },
+
+    ## getter methods ---------------
+
+    #' @description
+    #'
+    #' A method to extract a dataframe of simulated and/or observed data
+    #' (depending on instances of which classes have been added to the object).
+    #'
+    #' Note that the order in which you enter different object doesn't matter
+    #' because the returned dataframe is arranged alphabetically by dataset
+    #' name.
+    #'
+    #' @return
+    #'
+    #' In the returned dataframe, the following columns will always be present:
+    #'
+    #' name - group - dataType - xValues - xDimension - xUnit - yValues -
+    #' yErrorValues - yDimension - yUnit - yErrorType - yErrorUnit - molWeight
+
+
+    toDataFrame = function() {
+      # It is deliberate that this "cleaning" happens every time the user calls
+      # this method. R6 has a reference semantics, i.e., the object will be
+      # modified in place and the combined data will be updated with it. But,
+      # for future updates, we do need to retain some information that is
+      # omitted during cleaning.
+      #
+      # For example, we remove offset and scale factor columns during cleaning,
+      # but they need to be retained in the private copy of the combined
+      # dataframe because the datasets that might be added in the future will
+      # have their own offsets and scale factors, which will need to be appended
+      # to the existing ones to be returned by the `dataTransformations()`
+      # active binding. If we were to clean the dataframe in place, this will
+      # not be possible since there won't be offset and scale factors columns to
+      # append to.
+      private$.cleanDF(private$.dataCombined)
+    },
+
+    ## print method -----------------
+
+    #' @description
+    #' Print the object to the console
+
+    print = function() {
+      # group map contains names of the included datasets and grouping details
+      private$printLine("DataCombined", addTab = FALSE)
+      private$printLine("Datasets and groupings", addTab = FALSE)
+      cat("\n")
+      print(private$.groupMap)
+
+      # for method chaining
+      invisible(self)
+    }
+  ),
+
+  # active bindings ---------------------------------------------------
+
+  active = list(
+
+    #' @field names A vector of unique names of `DataSet` objects and/or
```

Comment:
"Unique names of data set contained in the `DataCombined` object" or something like this... We do not store `DataSet` or simulation results after adding.